bindings/java/native/src/classes/event_listener/event_manager.rs

```diff
@@ -0,0 +1,227 @@
+    }
+
+    pub fn subscribe_confirmation_state_change(
+        cb: Box<dyn TransactionConfirmationChangeListener + Send + 'static>,
+    ) -> EventId {
+        crate::block_on(async move {
+            iota_wallet::event::on_confirmation_state_change(move |event| {
+                cb.on_confirmation_state_change(event.clone());
+            })
+            .await
+        })
+    }
+
+    pub fn remove_confirmation_state_change_listener(event: EventId) {
+        crate::block_on(async move { iota_wallet::event::remove_confirmation_state_change_listener(&event).await })
+    }
+
+    pub fn subscribe_reattachment(cb: Box<dyn ReattachTransactionListener + Send + 'static>) -> EventId {
+        crate::block_on(async move {
+            iota_wallet::event::on_reattachment(move |event| {
+                cb.on_reattachment(event.clone());
+            })
+            .await
+        })
+    }
+
+    pub fn remove_reattachment_listener(event: EventId) {
+        crate::block_on(async move { iota_wallet::event::remove_reattachment_listener(&event).await })
+    }
+
+    pub fn subscribe_broadcast(cb: Box<dyn BroadcastTransactionListener + Send + 'static>) -> EventId {
+        crate::block_on(async move {
+            iota_wallet::event::on_broadcast(move |event| {
+                cb.on_broadcast(event.clone());
+            })
+            .await
+        })
+    }
+
+    pub fn remove_broadcast_listener(event: EventId) {
+        crate::block_on(async move { iota_wallet::event::remove_broadcast_listener(&event).await })
+    }
+
+    pub fn subscribe_transfer_progress(cb: Box<dyn TransferProgressListener + Send + 'static>) -> EventId {
+        crate::block_on(async move {
+            iota_wallet::event::on_transfer_progress(move |event| {
+                cb.on_transfer_progress(event.clone());
+            })
+            .await
+        })
+    }
+
+    pub fn remove_transfer_progress_listener(event: EventId) {
+        crate::block_on(async move { iota_wallet::event::remove_transfer_progress_listener(&event).await })
+    }
+
+    pub fn subscribe_balance_change(cb: Box<dyn BalanceChangeListener + Send + 'static>) -> EventId {
+        crate::block_on(async move {
+            iota_wallet::event::on_balance_change(move |event| {
+                cb.on_balance_change(event.clone());
+            })
+            .await
+        })
+    }
+
+    pub fn remove_balance_change_listener(event: EventId) {
+        crate::block_on(async move { iota_wallet::event::remove_balance_change_listener(&event).await })
+    }
+
+    pub fn subscribe_errors(cb: Box<dyn ErrorListener + Send + 'static>) -> EventId {
+        iota_wallet::event::on_error(move |error| {
+            cb.on_error(error.to_string());
+        })
+    }
+
+    pub fn remove_error_listener(event: EventId) {
+        iota_wallet::event::remove_error_listener(&event)
+    }
+
+    #[allow(unreachable_code)]
```

Comment:
is it possible to move the #[cfg] to the function name? or is the glue messed up in that case?

Reply:
Yea messed up, unless i make 2 cfg, one with (not any) and one with (any), which felt uglier